virttest/qemu_vm.py

```diff
@@ -458,28 +458,28 @@ def add_serial(devices, name, filename):
             if (not devices.has_option("chardev") or
                     not any(devices.has_device(dev)
                             for dev in ("isa-serial", "sclpconsole", "spapr-vty"))):
-                return " -serial unix:'%s',server,nowait" % filename
-
-            serial_id = "serial_id_%s" % name
-            cmd = " -chardev socket"
-            cmd += _add_option("id", serial_id)
-            cmd += _add_option("path", filename)
-            cmd += _add_option("server", "NO_EQUAL_STRING")
-            cmd += _add_option("nowait", "NO_EQUAL_STRING")
+                cmd = " -serial unix:'%s',server,nowait" % filename
+                return StrDev('SER-%s' % name, cmdline=cmd)
+            char_params = params.object_params(name)
```

Comment:
The `params` passed to `add_chardev()` (the `char_params` in this case) is not the same type as the `params` passed to `VM()`, you should build it by yourself.